File: Java/count-and-say.java
class Solution {
    public String countAndSay(int n) {  
        int i = 1;
        String result = "1";
        while (i < n) {
            result = getNext(result);
            i++;
        }
        return result;
    }
    public String getNext(String str) {
        char[] chars = str.toCharArray();
        int count = 1;
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < chars.length; i++) {
            if (chars[i - 1] == chars[i]) {
                count++;
            } else {
                sb.append(count + String.valueOf(chars[i - 1]));
                count = 1;
            }
        }
        sb.append(count + String.valueOf(chars[chars.length - 1]));
        return sb.toString();
    }
}

File: Kotlin/count-and-say.kt
class Solution {
    fun countAndSay(n: Int): String {
        var i = 1
        var result = ""
        while (i <= n) {
            result = getNext(result)
            i++
        }
        return result
    }
    fun getNext(str: String): String {
        var count = 1
        var result = ""
        for (i in 1 until str.length) {
            if (str[i] == str[i - 1]) {
                count++
            } else {
                result = result + count + str[i - 1]
                count = 1
            }
        }
        result = result + count + str.last()
        return result
    }
}
